exes/Main.hs

```diff
@@ -357,15 +357,19 @@ runAction opts = do
                -> return n
       _        -> fail $ "bad port number " ++ show str
 
+    checkHostURI :: ServerConfig -> Maybe String -> Int -> IO URI
     checkHostURI defaults Nothing port = do
       let guessURI       = confHostUri defaults
-          Just authority = uriAuthority guessURI
+      case uriAuthority guessURI of
+        Nothing -> fail "No URI Authority"
```

Comment:
this `fail` is the only change in this PR which might have some effect, is this acceptable?

Reply:
Why not simply disable the warning also in this case?
Is there evidence that this `Just` match can actually fail?
NB: Ideally unexpected failing matches would come with source location (`HasCallStack`).

Reply:
No problem, reverted the changes and applied the pragma, like the other files

It was the easiest of the warnings to fix, and using `fail` is consistent with the rest of the file IMO

There is no evidence the `Just` match can actually fail